Foundation/Thread.swift

```diff
@@ -130,37 +165,48 @@ open class Thread : NSObject {
             }
             ti = end_ut - CFGetSystemUptime()
         }
+#endif
     }
 
     open class func exit() {
         Thread.current._status = .finished
+#if os(Windows)
+        ExitThread(0)
+#else
         pthread_exit(nil)
+#endif
     }
 
     internal var _main: () -> Void = {}
-    private var _thread: pthread_t? = nil
+    private var _thread: _CFThreadRef? = nil
 
-#if CYGWIN
+#if os(Windows) && !CYGWIN
+    internal var _attr: _CFThreadAttributes =
+        _CFThreadAttributes(dwSizeOfAttributes: MemoryLayout<_CFThreadAttributes>.size,
+                            dwThreadStackReservation: 0)
+#elseif CYGWIN
     internal var _attr : pthread_attr_t? = nil
 #else
     internal var _attr = pthread_attr_t()
 #endif
     internal var _status = _NSThreadStatus.initialized
     internal var _cancelled = false
-    
+
     open private(set) var threadDictionary: NSMutableDictionary = NSMutableDictionary()
 
-    internal init(thread: pthread_t) {
-        // Note: even on Darwin this is a non-optional pthread_t; this is only used for valid threads, which are never null pointers.
+    internal init(thread: _CFThreadRef) {
```

Comment:
this might be the cause of the failure: because it is marked as internal?

Reply:
Well, it was previously `internal` ... we should be allowed to have internal interfaces that are unexposed right?

Reply:
it was previously the type of pthread_t which was from the Darwin module, where as _CFThreadRef pulls in the CoreFoundation module which has a naming conflict with the system CoreFoundation from my limited understanding.

Reply:
specifically I think this might be associated with things around inlining visibility.

Reply:
Hmm, I see how that got away in the past.  Suggestions on how to possibly address this then?

Reply:
there are two potential solutions here:

1) #ifdef the windows version to use the underlying type that _CFThreadRef is typedef'd from and leave the darwin version to be pthread_t
2) make this method fileprivate if possible.

Reply:
Another hackaround: you could move the typedef into the Swift module, and then check that it's the same type as _CFThreadRef.

Reply:
it seems that importing `typedef` is acting more like `newtype` than `typealias`.

Reply:
I *hope* it's not being treated as a CF type. That would be bad. But I *think* it's still a typealias; the compiler is just trying hard to preserve the sugar even though it doesn't need to.

Reply:
Okay, making it `fileprivate` is insufficient, its still failing on the import.  Does it make sense to actually `#if` check and desugar the conditional for Darwin and add a SR for ir?  Or just skip the SR entirely?

Reply:
@jrose-apple - can't just move the typedef as there are references to it in the C code.  I guess I could create a local typealias with a double underscore and see if that can get converted on the way down.

Reply:
You need to make something that *doesn't* reference the C one but is equivalent to it. Which stinks, but is probably the least bad answer.

Reply:
@jrose-apple - okay; think that this deserves a SR?

Reply:
Hmm, this is something that is odd, so I've created SR-9811.